verify-totp-sms/cypress/integration/tests.js

```diff
@@ -0,0 +1,66 @@
+describe('TOTP with SMS fallback integration tests', () => {
+  before(() => {
+    // TODO - add reset
+  });
+
+  it('Should load the index page', () => {
+    cy.visit('localhost:8000/index.html');
+    cy.get('#session-reset').click();
+    cy.get('#status').should('be.empty');
+    cy.get('#check-code').should('not.be.visible');
+
+    cy.get('#name').type('Taylor');
+    cy.get('[type="radio"]').check('whatsapp');
+    cy.get('#phone_number').type('201-555-0123');
+    cy.get('[type="submit"]').first().click();
+    cy.get('#status').should('be.visible');
+
+    cy.get('#status').should('not.have.css', 'color', 'rgb(169, 68, 66)'); // error color
+    cy.get('#status').contains('Sending verification code...');
+
+    cy.get('#check-code').should('be.visible');
+    cy.get('#code').type('1234');
+    cy.get('#check-code').submit();
+    cy.get('#modal-status').should('have.css', 'color', 'rgb(169, 68, 66)');
+    cy.get('#modal-status').contains('Incorrect token!');
```

Comment:
I'm pretty sure it's impossible to do an integration test with a successful verification here, any ideas or is what I have ok?

Reply:
You can tell Cypress to intercept calls to the backend and have it reply with what you want it to be. So you'd basically test the UI separately from the backend. That should also address your other question around environment variables because that way you don't actually need to have valid credentials. 

Here are some docs that are hopefully helpful. https://docs.cypress.io/api/commands/intercept#Matching-url

PS: thanks for digging into this :) I think this will make it so much easier to keep this maintained in the long run